clients/python/src/model_registry/_client.py

```diff
@@ -17,27 +19,55 @@ class ModelRegistry:
     def __init__(
         self,
         server_address: str,
-        port: int,
+        port: int = 443,
+        *,
         author: str,
-        client_key: str | None = None,
-        server_cert: str | None = None,
-        custom_ca: str | None = None,
+        user_token: bytes | None = None,
+        custom_ca: bytes | None = None,
     ):
         """Constructor.
 
         Args:
             server_address: Server address.
-            port: Server port.
+            port: Server port. Defaults to 443.
+
+        Keyword Args:
             author: Name of the author.
-            client_key: The PEM-encoded private key as a byte string.
-            server_cert: The PEM-encoded certificate as a byte string.
-            custom_ca: The PEM-encoded root certificates as a byte string.
+            user_token: The PEM-encoded user token as a byte string. Defaults to content of path on envvar KF_PIPELINES_SA_TOKEN_PATH.
+            custom_ca: The PEM-encoded root certificates as a byte string. Defaults to contents of path on envvar CERT.
         """
         # TODO: get args from env
         self._author = author
-        self._api = ModelRegistryAPIClient(
-            server_address, port, client_key, server_cert, custom_ca
-        )
+
+        if not user_token:
+            # /var/run/secrets/kubernetes.io/serviceaccount/token
+            sa_token = os.environ.get("KF_PIPELINES_SA_TOKEN_PATH")
+            if sa_token:
+                user_token = Path(sa_token).read_bytes()
+            else:
+                warn("User access token is missing", stacklevel=2)
+
+        root_certs = None
+        if not custom_ca:
+            ca_cert = os.environ.get("CERT")
+            if ca_cert:
+                root_certs = Path(ca_cert).read_bytes()
+            elif port == 443:
+                warn(
+                    "missing CA certificate, which is required for a secure connection",
+                    stacklevel=2,
+                )
+        else:
+            root_certs = custom_ca
+
+        if root_certs:
```

Comment:
For POSIX systems, there is a default list of certificate authorities as well. This could be a use case where user creates a server cert using a public CA. 
So, there is a third use case that's supported by [gRPC authentication](https://grpc.io/docs/guides/auth/#using-client-side-ssltls) where a custom CA is not specified, but an auth token is provided.